src/status_im/contexts/wallet/account/share_address/view.cljs

```diff
@@ -18,18 +17,18 @@
 
 (defn- share-action
   [address share-title]
-  (share/open
-   (if platform/ios?
-     {:activityItemSources [{:placeholderItem {:type    "text"
-                                               :content address}
-                             :item            {:default {:type "text"
-                                                         :content
-                                                         address}}
-                             :linkMetadata    {:title share-title}}]}
-     {:title     share-title
-      :subject   share-title
-      :message   address
-      :isNewTask true})))
+  (rf/dispatch [:open-share
+                (if platform/ios?
+                  {:activityItemSources [{:placeholderItem {:type    "text"
+                                                            :content address}
+                                          :item            {:default {:type "text"
+                                                                      :content
+                                                                      address}}
+                                          :linkMetadata    {:title share-title}}]}
+                  {:title     share-title
+                   :subject   share-title
+                   :message   address
+                   :isNewTask true})]))
```

Comment:
If the event expects two arguments, should we always pass them? IMO it's better to keep them in the same map if it means to pass an extra nil if there's no handlers.

Reply:
If we put them in same map, can we structure the map like this?
```
{:options ...
 :on-success ...
 :on-error ...}
```

Reply:
Maybe instead of `:options` we could use `:content`

Reply:
Works for me, but I think we should stick to :options, as you can pass more than just content there